src/vs/workbench/contrib/terminal/test/node/terminalProfiles.test.ts

```diff
@@ -0,0 +1,42 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as assert from 'assert';
+import { isWindows } from 'vs/base/common/platform';
+import { createStatProvider, detectAvailableShells } from 'vs/workbench/contrib/terminal/node/terminalProfiles';
+
+
+suite('Workbench - TerminalProfiles', () => {
+	suite('detectAvailableShells', () => {
+		if (isWindows) {
+			suite('detectAvailableWindowsShells', async () => {
+				test('should detect cmd prompt', async () => {
```

Comment:
Let's also add tests for:

- Non-existence (stat provider always returns false expected should be [])
- Multiple discovered shells
- Fallbacks

You should be able to make this relatively concise for these cases as well, eg:

```ts
test('git bash fallback', async () => {
	assert.deepStrictEqual(
		await detectAvailableShells(true, createStatProvider([
			`${process.env['ProgramW6432']}\\Git\\bin\\bash.exe`,
			`${process.env['LocalAppData']}\\Programs\\Git\\bin\\bash.exe`
		]));
		[{ profileName: 'Command Prompt', path: `${process.env['ProgramW6432']}\\Git\\bin\\bash.exe` }]
	);
	assert.deepStrictEqual(
		await detectAvailableShells(true, createStatProvider([
			`${process.env['ProgramW6432']}\\Git\\usr\\bin\\bash.exe`,
			`${process.env['LocalAppData']}\\Programs\\Git\\bin\\bash.exe`
		]));
		[{ profileName: 'Command Prompt', path: `${process.env['ProgramW6432']}\\Git\\usr\\bin\\bash.exe` }]
	);
	assert.deepStrictEqual(
		await detectAvailableShells(true, createStatProvider([
			`${process.env['ProgramFiles']}\\Git\\bin\\bash.exe`,
			`${process.env['LocalAppData']}\\Programs\\Git\\bin\\bash.exe`
		]));
		[{ profileName: 'Command Prompt', path: `${process.env['ProgramFiles']}\\Git\\bin\\bash.exe` }]
	);
	assert.deepStrictEqual(
		await detectAvailableShells(true, createStatProvider([
			`${process.env['ProgramFiles']}\\Git\\usr\\bin\\bash.exe`,
			`${process.env['LocalAppData']}\\Programs\\Git\\bin\\bash.exe`
		]));
		[{ profileName: 'Command Prompt', path: `${process.env['ProgramFiles']}\\Git\\usr\\bin\\bash.exe` }]
	);
	assert.deepStrictEqual(
		await detectAvailableShells(true, createStatProvider([
			`${process.env['LocalAppData']}\\Programs\\Git\\bin\\bash.exe`
		]));
		[{ profileName: 'Command Prompt', path: `${process.env['LocalAppData']}\\Programs\\Git\\bin\\bash.exe` }]
	);
});
```

Reply:
Deferred: https://github.com/microsoft/vscode/issues/118997